src/Appwrite/Domain/Validator/AppwriteDomain.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Appwrite\Domain\Validator;
+
+use Utopia\System\System;
+use Utopia\Validator\Domain;
+
+class AppwriteDomain extends Domain
```

Comment:
This validator sounds too complex.

Some things sound related to domains in general, not our subdomain issue. For that, inside isValid, we could run parent::isValid(), to run standard domain's validation.

What I would expect in this new validator is just a constructor with integer of amount of domains allowed (top-level domain, domain, subdomain, sub-sub domain...). And an isValid that splits by dot and compares count.
Due to how simple this it, we might want to add this to original Domain validator, without need to make a new one - just adding 1 if check on top of what's already there, and to make amount of subdomians configurable, add parameter to constructor. From Appwrite we can then set `domains: 3`, to only allow to have 3 parts like `myapp.appwrite.network`.